extensions/s3/src/main/java/io/deephaven/extensions/s3/S3ChannelContext.java

```diff
@@ -303,29 +304,29 @@ public boolean equals(Object obj) {
         private final long from;
         private final long to;
         private final Instant createdAt;
-        private CompletableFuture<ByteBuffer> consumerFuture;
-        private volatile CompletableFuture<ByteBuffer> producerFuture;
+        private volatile CompletableFuture<Boolean> consumerFuture;
```

Comment:
I don't understand (yet) what part of the change necessitates that this reference be `volatile`.